plugin-k8s/src/cgroup_v2.rs

```diff
@@ -104,9 +128,172 @@ pub fn gather_value(file: &mut CgroupV2MetricFile, content_buffer: &mut String)
+    headers.insert(
+        header::AUTHORIZATION,
+        format!("Bearer {}", token)
+            .parse()
+            .expect("Can't parse formatted token into a HeaderName"),
+    );
+    let client = reqwest::Client::builder()
+        .danger_accept_invalid_certs(true) 
+        .default_headers(headers)
+        .build().unwrap();
+    let response_raw = client.get(api_url).send().await;
+
+    let response = match response_raw {
+        Ok(resp) => {
+            resp
+        },
+        Err(_) => {
+            return HashMap::new();
+        }
+    };
+
+    let data: Value = response.json().await.expect("Error parsing JSON");
+    let mut hash_map_to_ret = HashMap::new();
+
+    if let Some(items) = data.get("items") {
+        for item in items.as_array().unwrap_or(&vec![]) {
+            let metadata = item.get("metadata").unwrap_or(&Value::Null);
+            let spec = item.get("spec").unwrap_or(&Value::Null);
+            let annotations = metadata.get("annotations").unwrap_or(&Value::Null);
+            let mut config_hash = annotations
+                .get("kubernetes.io/config.hash")
+                .and_then(|v| v.as_str())
+                .unwrap_or("");
+            if config_hash == "" {
+                match metadata {
+                    Value::Null => {
+                        continue;
+                    }
+                    _ => {
+                        config_hash = metadata.get("uid")
+                        .and_then(|v| v.as_str())
+                        .unwrap_or("");
+                    }
+                }
+            }
+           
+            let pod_name = metadata.get("name").and_then(|v| v.as_str()).unwrap_or("");
+            let pod_namespace = metadata.get("namespace").and_then(|v| v.as_str()).unwrap_or("");
+            let pod_node = spec.get("nodeName").and_then(|v| v.as_str()).unwrap_or("");
+            log::debug!("Found matching pod: {} in namespace {}", pod_name, pod_namespace);
+            hash_map_to_ret.entry(String::from(config_hash)).or_insert((pod_name.to_owned(), pod_namespace.to_owned(), pod_node.to_owned()));
+            
+        }
+    } else {
+        log::debug!("No items found in the JSON response.");
+    }
+
+    return hash_map_to_ret;
+}
+
+
+//Read files in a filesystem to associate a cgroup of a poduid to a kubernetes pod name
+pub async fn get_pod_name(uid: String) -> (String, String, String) {
+    let new_uid = uid.replace("_", "-");
+    let output = Command::new("kubectl")
+        .args(&["create", "token", "alumet-reader"])
+        .output();
+    
+    let output_unwraped = match output {
+        Err(_) => {
+            return ("".to_string(),"".to_string(),"".to_string());
+        },
+        Ok(output_tmp) => {
+            output_tmp
+        }
+    };
+
+    let token = String::from_utf8_lossy(&output_unwraped.stdout);
+    let token = token.trim();
+    let api_url = "https://10.22.80.14:6443/api/v1/pods/";
```

Comment:
Is it possible to only request the pod you want ?

Reply:
curl -X GET $APISERVER/api/v1/pods?fieldSelector=spec.nodeName=NODE_NAME--header "Authorization: Bearer $TOKEN" --insecure

Yes it is possible but, how to know what the node name is without adding a query? Indeed the hostname and the name of the kubernetes node can be different

Reply:
Well, you do have the pod's UID if I' not mistaken

Reply:
?fieldSelector=spec.nodeName=NODE_NAME is filtering the pods running on the node, not only the pod you want

Reply:
well sorry, misunderstood, i was about requesting only the node we are on. 
As i can't see, i can't use a fiel selector on the pod uid as available field selector are:

- spec.nodeName
- spec.restartPolicy
- spec.schedulerName
- spec.serviceAccountName
- spec.hostNetwork
- status.phase
- status.podIP
- status.nominatedNodeName

Source [kubernetes.io](https://kubernetes.io/docs/concepts/overview/working-with-objects/field-selectors/)

Reply:
Oh yes my bad... I thought such a basic feature was already present in k8s
Well, you can restrict the request on the nodeName at least

Reply:
Done